test/node/bson_test.js

```diff
@@ -38,22 +38,13 @@ BSON.BSON_BINARY_SUBTYPE_UUID = 3;
 BSON.BSON_BINARY_SUBTYPE_MD5 = 4;
 BSON.BSON_BINARY_SUBTYPE_USER_DEFINED = 128;
 
-// var hexStringToBinary = function(string) {
-//   var numberofValues = string.length / 2;
-//   var array = '';
-
-//   for (var i = 0; i < numberofValues; i++) {
-//     array += String.fromCharCode(parseInt(string[i * 2] + string[i * 2 + 1], 16));
-//   }
-//   return array;
-// };
-
-var assertBuffersEqual = function(test, buffer1, buffer2) {
-  if (buffer1.length !== buffer2.length)
-    test.fail('Buffers do not have the same length', buffer1, buffer2);
+var assertBuffersEqual = function(done, buffer1, buffer2) {
```

Comment:
someday we should probably switch all the times this function is used to just use expect(buffer1).to.deep.equal(buffer2). unless if this is doing something different I'm not seeing?

Reply:
Also correct.  Chai used to not do deep comparison between buffers, so I believe that's why we ended up with this helper. (It predates even Buffer.compare!)